Language.md

```diff
@@ -1 +1,74 @@
+
+But the important thing here is that the program doesn't have to **wait** for the response.
+It can do other important things and deal with the outcome when it is ready to grace us with it's presence.
+
+This has unbelievable advantages(and disadvantages,*callback hell is a terrible place to be in ,more on that later*)
+
+* The primary advantage is increased responsiveness and the illusion of speed, which is very important for a web developer.
+
+* The second advantage can be engaging in long lasting database interactions(background tasks)
+
+Async support is uncommmon in a lot of languages (though it is picking up steam),JavaScript is pretty ahead of the game on this front.
+
+**4. Server Side A.K.A Node.js**
+
+In the early days JavaScript was really limited, which means it was only useful on the client side. The recent phenomenon of node.js has changed this scenario rapidly. Node.js can frankly add a backend to your frontend using only JavaScript, Another way of looking at it is ,it can create a server.
+No PHP, No Python needed.
+
+This has led to applications being created only in one language. So we can interact with the database, handle client requests all using Javascript.
+
+**5. Animations fast and fluid (Client Side Interaction)**
+
+This language was always a client side affair first. It's adoption has led to support for all the beautiful animations ,parallax effects and overall responsiveness.
+Libraries like JQuery provide great looking effects in a very easy to learn package.
+Some quick features are:
+
+* Manipulate the DOM like a boss
+* Send Https requests
+* Animations
+
+
+**6. It pays to learn JavaScript**
+
+Companies are looking for good JavaScript devs. It is a useful skill and if learnt properly can give you a career. JavaScript devs are plentiful but good JavaScript devs are rare and companies shell out a fair penny for people who know their way around the workings of this mysterious langague.
+
+**7. Dynamic Browser Side Rendering**
+
+Frameworks such as Angular, React and Meteor are changing the way client side coding is done. The Model View Container(MVC) principle has been taken to heart and now it is possible to *bind* an element such that change can be reflected instantaneously(Think the way a Facebook like changes when clicked upon).
+
+
+New and better ways of rendering html elements have resulted in cleaner and faster code. Now it is possible to render an entirely new html page in a div.
+This has given rise to a new breed of websites know as Single Page Applications, **S.P.A**. HackerRank is a prime example. The URL changes but the site never really reloads
+
+ Companies like Google and Facebook are at the forefront of these technologies and you know that those guys know their stuff.
+
+
+ I hope this post convinces you to learn JavaScript. It really is a great language and totally worth your time.
+
```

Comment:
Now mention about React Native, Cordova/Ionic that provide JS tech stack for Mobile App Development. React Native actually allows you to build native applications using principles of ReactJS for both iOS and Android. This not only increases code-reuse but also allows a JS Dev to use their favourite modules over NPM.